Marlin/SanityCheck.h

```diff
@@ -292,7 +292,15 @@
   #endif
 
   /**
+   * Check if ENDSTOPS_ONLY_FOR_HOMING is disabled when M211 command is enabled
+   */
+  #if (defined(ENABLE_M211)) && (!defined(ENDSTOPS_ONLY_FOR_HOMING))
+    #error ENDSTOPS_ONLY_FOR_HOMING must be enabled when M211 command is active.
+  #endif
+
+  /**
    * Warnings for old configurations
+   * Check if ENDSTOPS_ONLY_FOR_HOMING is disabled when M211 command is enabled
```

Comment:
Or you can just set Z to 10 by G92 Z10. If it's at zero you can now go down by 10mm.